infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/DefaultClusterManager.java

```diff
@@ -182,179 +202,322 @@ public TopologyDescriptor readConfig() {
+                    Integer.parseInt(sinkCorfuPorts.get(i)));
 
             for (int j = 0; j < sinkNodeNames.size(); j++) {
-                log.info("Sink Cluster Name {}, IpAddress {}",
-                    sinkNodeNames.get(j), sinkNodeHosts.get(i));
-                NodeDescriptor nodeInfo =
-                    new NodeDescriptor(sinkNodeHosts.get(i),
-                        sinkLogReplicationPorts.get(i), SINK_CLUSTER_NAME,
-                        sinkNodeIds.get(i), sinkNodeIds.get(i));
+                NodeDescriptor nodeInfo = new NodeDescriptor(sinkNodeHosts.get(i), sinkLogReplicationPorts.get(i),
+                        sinkClusterIds.get(i), sinkNodeIds.get(i), sinkNodeIds.get(i));
                 sinkCluster.getNodesDescriptors().add(nodeInfo);
             }
-            sinkClusters.add(sinkCluster);
+            sinkClustersToReplicationModel.put(sinkCluster,
+                    new HashSet<LogReplication.ReplicationModel>(){
+                        {
+                            add(LogReplication.ReplicationModel.FULL_TABLE);
+                        }});
+            allClusters.put(sinkCluster.getClusterId(), sinkCluster);
         }
 
-        return new TopologyDescriptor(0L, sourceClusters, sinkClusters, localEndpoint);
-    }
-
-    private TopologyConfigurationMsg constructTopologyConfigMsg() {
-        TopologyDescriptor clusterTopologyDescriptor = readConfig();
-        return clusterTopologyDescriptor.convertToMessage();
+        return new TopologyDescriptor(0L, sinkClustersToReplicationModel, sourceClustersToReplicationModel,
+                allClusters, new HashSet<>(), localNodeId);
     }
 
 
     @Override
-    public void register(CorfuReplicationDiscoveryService corfuReplicationDiscoveryService) {
-        this.corfuReplicationDiscoveryService = corfuReplicationDiscoveryService;
+    public TopologyDescriptor queryTopologyConfig(boolean useCached){
+        try {
+            log.info("Wait until a topology is created");
+            waitForTopologyInit.await();
+        } catch (InterruptedException e) {
+            log.error("Interrupted");
+        }
+        log.debug("new cluster config msg " + topologyConfig);
+        return topologyConfig;
     }
 
-    @Override
-    public void setLocalEndpoint(String endpoint) {
-        this.localEndpoint = endpoint;
+    private void createSingleSourceSinkTopologyBuckets() {
+        topologyConfig = readConfig();
+
+        Map<ClusterDescriptor, Set<LogReplication.ReplicationModel>> remoteSourceToReplicationModels = new HashMap<>();
+        Map<ClusterDescriptor, Set<LogReplication.ReplicationModel>> remoteSinkToReplicationModels = new HashMap<>();
+        Set<ClusterDescriptor> connectionEndPoints = new HashSet<>();
+
+        ClusterDescriptor localCluster = findLocalCluster();
+
+        if(localCluster.getRole().equals(ClusterRole.SOURCE)) {
```

Comment:
shouldnt this check stop being used with this change?

Reply:
No, since this file is majorly for testing purpose, we can simulate different kinds of topology, but we'll need to use SOURCE and SINK.